BasicShaping.roboFontExt/lib/arabicFeatures.py

```diff
@@ -0,0 +1,70 @@
+from fontTools.feaLib.parser import Parser
+from fontTools.feaLib.ast import *
+from ufo2ft.featureWriters.ast import getScriptLanguageSystems, findFeatureTags
+
+SUFFIX_FEA_MAP = {  # suffix: feature_tag
+                'init': 'init',
+                'medi': 'medi',
+                'fina': 'fina',
+                }
+
+def _getDefaultFeaFile():
+    feaFileAst = FeatureFile()
+    lang = LanguageSystemStatement("DFLT", "dftl")
+    feaFileAst.statements.append(lang)
+    return feaFileAst
+
+def _getFeatureFileAST(font):
+    featurefile = UnicodeIO(font.features.text)
+    featurefile.name = font.path
+    parser = Parser(featurefile, font.keys())
+    return parser.parse()
+
+def getAstIndexMap(feaFile):
+    """
+    Returns a dict in which the keys are feaLib.ast class and value is a list of
+    indices which represents where is the object in feature file statements
+    list.
+    """
+    indexMap = {}
+    for i, statement in enumerate(feaFile.statements):
+        indexMap.setdefault(type(statement), set()).add(i)
+    return indexMap
+
+def getFeatures(font):
+    feaFileAst = _getDefaultFeaFile()
+    if font.features.text is not None and font.features.text.strip():
+        try:
+            feaFileAst = _getFeatureFileAST(font)
+        except FeatureLibError as e:
+            print(e)
+    langSysMap = getScriptLanguageSystems(feaFileAst)
+    astIndexMap = getAstIndexMap(feaFileAst)
+    if "Arab" not in langSysMap:
+        arabLang = LanguageSystemStatement("Arab", "dftl")
+        index = max(astIndexMap.get(LanguageSystemStatement, 0)) + 1
```

Comment:
`max` takes a list this is now a single index, this raises an error when no languagesystem is set in the fea  code

Reply:
Oops! I fixed it.